test/Microsoft.AspNetCore.SignalR.Client.Tests/TestConnection.cs

```diff
@@ -116,9 +117,26 @@ public Task ReceiveBytesAsync(byte[] bytes)
         {
             return Application.Output.WriteAsync(bytes).AsTask();
         }
-        public async Task<string> ReadSentTextMessageAsync()
+
+        public async Task<string> ReadSentTextMessageAsync(bool ignorePings = true)
         {
             // Read a single text message from the Application Input pipe
+
+            while (true)
+            {
+                var result = await ReadSentTextMessageAsyncInner();
+
+                var receivedMessageType = (int)JObject.Parse(result).Property("type").Value;
```

Comment:
You can get even fancier 😸 . `JObject` can be accessed like a dictionary: `(int)JObject.Parse(result)["type"]`